File: Simple Factory/JavaScript/NotificationFactory.js
const EmailNotification = require('./EmailNotification');
const SmsNotification = require('./SmsNotification');
const PushNotification = require('./PushNotification');

class NotificationFactory {

    static createNotification(type) {

        switch (type.toLowerCase()) {
            case "email":
                return new EmailNotification();
            case "sms":
                return new SmsNotification();
            case "push":
                return new PushNotification();
            default:
                throw new Error("Invalid notification type");
        }
    }
}

module.exports = NotificationFactory;

File: Simple Factory/C#/SimpleFactoryApp/Notifications/NotificationFactory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using System;

namespace SimpleFactoryApp.Notifications
{
    public class NotificationFactory
    {
        public static INotification CreateNotification(string type)
        {
            switch (type.ToLower())
            {
                case "email":
                    return new EmailNotification();
                case "sms":
                    return new SmsNotification();
                case "push":
                    return new PushNotification();
                default:
                    throw new ArgumentException("Invalid notification type");
            }
        }
    }
}
